src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs

```diff
@@ -0,0 +1,227 @@
+{
+    public class AtlassianService : IAtlassianService
+    {
+        private readonly IApiClient _apiClient;
+        private readonly ILogger<AtlassianService> _log;
+        private readonly OnBoardingToolFunctions _config;
+        private readonly AbstractValidator<Value> _validator;
+        
+        public AtlassianService(
+            IApiClient apiClient,
+            ILogger<AtlassianService> log,
+            IOptions<OnBoardingToolFunctions> config,
+            AbstractValidator<Value> validator)
+        {
+            _config = config.Value;
+            _apiClient = apiClient;
+            _log = log;         
+            _validator = validator;
+        }
+
+        public async Task ProcessCreateUserRequests()
+        {
+            _log.LogInformation(_config.ToString());
+            _log.LogInformation($"Processing create user requests");
+
+            var requests = await GetUserRequests();
+
+            if(requests.Size == 0)
+            {
+                _log.LogInformation($"No create user requests to process");
+                return;
+            }
+
+            var tasks = new List<Task>();
+
+            foreach (var request in requests.Values
+                .AsEnumerable()
+                .Where(r => r.Fields.Customfield_12105.Value.Equals("grant", StringComparison.InvariantCultureIgnoreCase)))
+            {                
+                var email = request.Fields.Customfield_12104;
+
+                _log.LogInformation($"Processing request for email {email} from reporter {request.Fields.Reporter.DisplayName}");
+
+                var result = _validator.Validate(request, opts => opts.IncludeRuleSets(ValueValidator.CanCreate));
+                if(!result.IsValid)
+                {
+                    _log.LogInformation($"Request failed validation {string.Join(',', result.Errors.Select(err => err.ErrorMessage).ToArray())}");
+                    continue;
+                }
+                    
+                tasks.Add(CreateUser(request, new User { Username = email,  Email = email })
+                .ContinueWith(TransitionRequest, TaskContinuationOptions.OnlyOnRanToCompletion));
+            }
+
+            Task.WaitAll(tasks.ToArray());
+
+            _log.LogInformation($"Finished processing create requests");
+        }
+
+        public async Task ProcessRevokeUserRequests()
+        {
+            _log.LogInformation(_config.ToString());
+            _log.LogInformation($"Processing revoke user requests");
+
+            var requests = await GetUserRequests();
+
+            if(requests.Size == 0)
+            {
+                _log.LogInformation($"No revoke user requests to process");
+                return;
+            }
+
+            var tasks = new List<Task>();
+
+            foreach (var request in requests.Values
+                .AsEnumerable()
+                .Where(r => r.Fields.Customfield_12105.Value.Equals("revoke", StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Your MMV, I prefer the coding style of declaring the list up front, 

```
var revokeRequests = requests.ValuesAsEnumerable()Where(r => r.Fields.Customfield_12105.Value.Equals("revoke", StringComparison.InvariantCultureIgnoreCase))`
                
                foreach(var request in revokeRequests)
```
                
As I think you get a neater declaration, and descriptive variable of the held list. This does give a negligible performance benefit but it's hardly worth mentioning as a for or against as my main item is the name and command separation